azext_edge/tests/edge/support/conftest.py

```diff
@@ -31,17 +31,21 @@ def add_pod_to_mocked_pods(
         spec = V1PodSpec(containers=[V1Container(name=container_name)])
         pod = V1Pod(metadata=V1ObjectMeta(namespace=namespace, name=pod_name), spec=spec)
 
-        if mock_init_containers:
+        if mock_init_container:
             pod.spec.init_containers = [V1Container(name="mock-init-container")]
-            expected_pod_map[namespace][pod_name]["mock-init-container"] = mock_log
+            expected_pod_map[namespace][pod_name] = {"mock-init-container": mock_log}
         pod_list.append(pod)
-        expected_pod_map[namespace][pod_name] = {container_name: mock_log}
+
+        if pod_name in expected_pod_map[namespace]:
+            expected_pod_map[namespace][pod_name].update({container_name: mock_log})
```

Comment:
if it is only one key, then use the [] notation. Update is more for updating multiple keys in a dict

Reply:
I see, will update!